x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_security_solution_navigation.test.tsx

```diff
@@ -29,23 +34,57 @@ describe('Security Solution Navigation', () => {
   beforeEach(() => {
     jest.clearAllMocks();
   });
+  describe('when classic navigation is enabled', () => {
+    beforeAll(() => {
+      mockGetChromeStyle$.mockReturnValue(of('classic'));
+    });
 
-  it('should return proper navigation props', () => {
-    const { result } = renderHook(useSecuritySolutionNavigation);
-    expect(result.current).toEqual(
-      expect.objectContaining({
+    it('should return proper navigation props', async () => {
+      const { result } = renderHook(useSecuritySolutionNavigation);
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          canBeCollapsed: true,
+          name: 'Security',
+          icon: 'logoSecurity',
+          closeFlyoutButtonPosition: 'inside',
+        })
+      );
+
+      // check regular props
+      expect(result.current).toEqual({
         canBeCollapsed: true,
         name: 'Security',
         icon: 'logoSecurity',
+        children: expect.anything(),
```

Comment:
I'd recommend to mock  `SecuritySideNav` from '../security_side_nav' to expect an explicit value here

Reply:
I've tried to do this, the `SecuritySideNav` component is already mocked, but `children` is a _ReactNode_, and `isEqual` crashes when trying to assert it. So I decided to check only the regular props here, making sure the `children` exists, and then I added an assertion for the `children`, starting at line 63:

```
      // check rendering of SecuritySideNav children
      const { findByTestId } = render(<>{result.current?.children}</>);

      expect(mockSecuritySideNav).toHaveBeenCalled();
      expect(await findByTestId('SecuritySideNav')).toBeInTheDocument();
```

If you know any better way to do this please let me know.